src/debuggability.js

```diff
@@ -764,12 +821,7 @@ var fireGlobalEvent = (function() {
             }
         };
     } else {
-        var globalObject = typeof self !== "undefined" ? self :
```

Comment:
Agree. There is no reason to worry at all, because the definition is in global scope of utils.js file, library users that write "self = this" will not affect this. The only case when user overrides self object in global scope should not be handled in my opinion.